src/Util/HexString.hpp

```diff
@@ -69,4 +69,22 @@ ParseHexString(const std::string_view hex_str)
 
   return raw_hash;
 }
+
+template<std::size_t len>
+std::string
```

Comment:
Using `std::string` here is unnecessary bloat. You know the size already at compile time; just return `StringBuffer<len*2+1>`!